apps/api/pages/api/teams/_post.ts

```diff
@@ -91,8 +122,26 @@ async function postHandler(req: NextApiRequest) {
   return {
     team: schemaTeamReadPublic.parse(team),
     owner: schemaMembershipPublic.parse(team.members[0]),
-    message: "Team created successfully, we also made you the owner of this team",
+    message: isAdmin
+      ? "Team created successfully, we also made user with submitted userId the owner of this team"
+      : "Team created successfully, we also made you the owner of this team",
   };
 }
 
+async function checkPermissions(req: NextApiRequest) {
+  const { isAdmin } = req;
+  const body = schemaTeamCreateBodyParams.parse(req.body);
+
+  /* Non-admin users can only create teams for themselves */
+  if (!isAdmin && body.ownerId)
+    throw new HttpError({
+      statusCode: 401,
+      message: "ADMIN required for `ownerId`",
+    });
+
+  /* Admin users are required to pass in a userId */
+  if (isAdmin && !body.ownerId)
+    throw new HttpError({ statusCode: 400, message: "`ownerId` required for ADMIN" });
```

Comment:
@zomars with the recent changes that went in #12501 do we want more checks in here to check if an active subscription exists? It would be easier to allow them creating teams and leave them unpublished by default, but I am not sure which option would be the best here.

Reply:
This is a good point. This could allow users from our hosted version to bypass the stripe checkout needed to create a team. IMO we could allow to create a new team and return the checkout URL in the api response. Then the user would need to finish the checkout in a browser and on the callback finish the team creation. Does it make sense?

Reply:
Yeah, sounds good. The team would then only be publishable after the browser checkout succeeds and team creation finishes.
cc: @supalarry

Reply:
Closed this PR and opened new one ready for review:
 https://github.com/calcom/cal.com/pull/13211